### ISSUE TITLE
SnappyMail `autocreate_system_folders`

### ISSUE THREAD

**Issue by usmannasir** [the-djmaze/snappymail#907]
**Describe the bug**
Does the latest version of SnappyMail not support ? `autocreate_system_folders`

**To Reproduce**

I am CyberPanel developer and we are facing this problem -> https://community.cyberpanel.net/t/snappymail-does-not-show-email-folders-for-subdomain-domain-com-websites/40026

**Expected behavior**
SnappyMail to take value of ``autocreate_system_folders` from application.ini

**Screenshots**

**Please complete the following information:**
- Browser: [e.g. chromium 100, firefox 100, safari 14, mobile]
- IMAP daemon: [e.g. courier, dovecot]
- PHP version:
- SnappyMail Version:

**[Debug/logging information](https://github.com/the-djmaze/snappymail/wiki/FAQ#how-do-i-enable-logging)**
Place them here (few lines) or as attachments (many lines)

**Additional context**

**Comment by the-djmaze** [comment]
It is moved to a plugin.
https://github.com/the-djmaze/snappymail/tree/master/plugins/mailbox-detect

**Comment by usmannasir** [comment]
Oh, thanks for the quick response.

Is there a way to activate and install this by default via code? (I mean not from interface).

**Comment by the-djmaze** [comment]
Yes, you can write a PHP script to do that.

Nextcloud integration does the same at install.
See https://github.com/the-djmaze/snappymail/blob/master/integrations/nextcloud/snappymail/lib/Migration/InstallStep.php

**Comment by usmannasir** [comment]
Many thanks, I managed to fix this by writing a peice in python https://github.com/usmannasir/cyberpanel/commit/f7e4f5d91cb2b40473ae6ceb4ca473493ab3bb0c

I will be closing this now, however, I believe this feature should not be offered as plugin, very core thing.

thanks again.

**Comment by usmannasir** [comment]
PS: I've managed to install and activate plugin via code and it also shows enabled in the admin panel as well, but it wont create system folders unless i manually go to admin panel and disable/enable the plugin.

Any pointers?

**Comment by the-djmaze** [comment]
In the config directory where application.ini is, there should be a mailbox-detect.json file.
You can create one at install, just use your current as default.

**Comment by usmannasir** [comment]
Yeah added that too, it seems we also need to manually create `application.ini` as well, because on fresh install this file don't get created automatically.

**Comment by the-djmaze** [comment]
> Yeah added that too, it seems we also need to manually create `application.ini` as well, because on fresh install this file don't get created automatically.

True, that's why i mentioned the Nextcloud script.
If you just created a PHP script and run `php script.php` from Python or whatever else (as the sudo user who will own the files) then application.ini gets created and you can install the package and do more.

Now you are doing it outside SnappyMail and whenever i remove/add/change options to the application.ini, you must modify your python scripts.

**Comment by usmannasir** [comment]
You mean If I run this file `https://github.com/the-djmaze/snappymail/blob/master/integrations/nextcloud/snappymail/lib/Migration/InstallStep.php` via PHP CLI, application.ini will be generated? also where should i place this file?

**Comment by the-djmaze** [comment]
No, that script is for reference. I will write one for youe 😉

**Comment by usmannasir** [comment]
oh ok, thanks alot

The issue I am facing is that I am trying to edit application.ini after unpacking SnappyMail, but ofcourse it does not exists.

Also we don't use standard directory structure, if you need any help you can email me regarding that.

**Comment by the-djmaze** [comment]
Here you go https://github.com/the-djmaze/snappymail/tree/master/integrations/cyberpanel

**Comment by usmannasir** [comment]
Should I place this file under `/usr/local/CyberCP/public/snappymail` and just run it?

**Comment by the-djmaze** [comment]
You can place it anywhere you want and outside public would be best, else anyone can call the script multiple times.
(i could make it detect CLI and exit of not called from command line)

I haven't tested it, and you can make the file -x executable (see first line in script).

**Comment by usmannasir** [comment]
Let me see if it works, i will revert

**Comment by usmannasir** [comment]
So far looks like it is working, much thanks

**Comment by the-djmaze** [comment]
Great!

Also instead of https://github.com/usmannasir/cyberpanel/blob/f7e4f5d91cb2b40473ae6ceb4ca473493ab3bb0c/install/install.py#L1267

You could fetch https://snappymail.eu/repository/latest.tar.gz

**Comment by usmannasir** [comment]
I will check into this, however bundle of thanks

**Comment by usmannasir** [comment]
It looks like people are still having the issues https://community.cyberpanel.net/t/snappy-mail-auto-create-folder-issue-not-solved-after-update/40232

**Comment by the-djmaze** [comment]
Then i will reopen this issue

**Comment by the-djmaze** [comment]
When creation of folder fails, it writes to the log.
But the log is disabled by default.

To get as much info as needed:
1. enable debug mode (application.ini or ?admin#/config)
2. Open the mailbox
3. Read the log

See https://github.com/the-djmaze/snappymail/wiki/FAQ#how-do-i-enable-logging

**Comment by usmannasir** [comment]
Someone reported this 
```
yes, I have tried to run that run PHP script manually but that’s a temporary solution, folders are getting created for existing mail while creating new mail issue is repeating, and I can’t see any mail in the inbox
```
For my test folders are being created as well. But as per reports emails are not landing in inbox.

**Comment by the-djmaze** [comment]
When send it should go to the "sent" box, not the inbox.

Check: User -> Settings -> Folders -> [System Folders]
![afbeelding](https://user-images.githubusercontent.com/3752035/217302110-2dc52144-e755-43ef-966d-2ae4df617118.png)

![afbeelding](https://user-images.githubusercontent.com/3752035/217302224-665b105d-7e22-4a18-a547-78c1d5120f4e.png)

**Comment by usmannasir** [comment]
I believe what this guy was saying is that when he tries to send email to the server, he won't see the email in inbox.

Although I've asked him if he set his MX records.

Ref https://community.cyberpanel.net/t/snappy-mail-auto-create-folder-issue-not-solved-after-update/40232/10?u=usmannasir

**Comment by usmannasir** [comment]
@the-djmaze We've supported CyberPanel on arm and folders are not getting generated, any idea?

thanks.

**Comment by usmannasir** [comment]
Getting following error on the file you wrote for CyberPanel 

```
/usr/local/lsws/lsphp74/bin/php /usr/local/CyberCP/snappymail_cyberpanel.php
PHP Notice:  Invalid plugin class MailboxDetectPlugin in /usr/local/CyberCP/public/snappymail/snappymail/v/2.25.3/app/libraries/RainLoop/Plugins/Manager.php on line 152
```
